src/app/clusters/ota-requestor/GenericOTARequestorDriver.cpp

```diff
@@ -376,6 +382,7 @@ bool GenericOTARequestorDriver::GetNextProviderLocation(ProviderLocationType & p
     if (iterator.Next())
     {
         providerLocation = iterator.GetValue();
+        listExhausted    = true;
```

Comment:
Is `listExhausted` meant to represent we've reached the end of the list or that we've wrapped all the way around? For instance, if there are 5 providers in the list and the `lastUsedProvider` is the fifth one, we would mark `listExhausted = true` with the current logic but is it meant to only mark if we've wrapped around.e.g. gotten through providers 1-4 and gotten back to the fifth one to say list has been exhausted?